### ISSUE TITLE
Clean up tandems in Sim

### ISSUE THREAD

**Issue by samreid** [phetsims/joist#446]
Some places are using `+'.sim.Something` and some places do not even need `sim` at all.  We should look through these tandems and clean them up.  Closely related to  #445.

**Comment by zepumph** [comment]
@samreid please review

**Comment by samreid** [comment]
I'm wondering if we can eliminate `simTandem` and just use `tandem`.  Also, can the `barrierRectangle` be uninstrumented?

**Comment by zepumph** [comment]
>I'm wondering if we can eliminate simTandem and just use tandem

That isn't a bad idea. I kinda like how it is organized, but maybe it just ends up being more confusing. 

> Also, can the barrierRectangle be uninstrumented?

We have gone down this road before. If I remember correctly the answer is no, because in the data stream, I think it is the only way to know that a dialog closed (or how it was closed) if you click away from the dialog. Currently we have it set up as `dataStreamOnlyType`. Once `phetioMethods` is implemented, I was thinking that will turn into `{ phetioState: false, phetioMethods: false }`.

**Comment by samreid** [comment]
> We have gone down this road before. If I remember correctly the answer is no, because in the data stream, I think it is the only way to know that a dialog closed (or how it was closed) if you click away from the dialog.

Scenery has a new way to detect when clicking away from a dialog, perhaps we can use that instead.

**Comment by samreid** [comment]
@jonathanolson do you think we can eliminate joist's barrier rectangles and use Display.addInputListener instead?

**Comment by jonathanolson** [comment]
> @jonathanolson do you think we can eliminate joist's barrier rectangles and use Display.addInputListener instead?

Presumably not, it notifies the display after dispatching the event to nodes.

**Comment by zepumph** [comment]
>I think it is done by adding a listener to the display.

Is this what you are thinking of? Example from ESPB ControPointUI.js:
```js
 this.clickToDismissListener = {
      down: function() {
        if ( enableClickToDismissListener ) {
          if ( _.indexOf( phet.joist.display.getInputListeners(), self.clickToDismissListener ) !== -1 ) {
            phet.joist.display.removeInputListener( self.clickToDismissListener );
          }
          self.detach();
        }
        else {
          enableClickToDismissListener = true;
        }
      }
    };
    phet.joist.display.addInputListener( this.clickToDismissListener );

**Comment by zepumph** [comment]
I reverted the above commit because currently there are a lot of places in the wrapper code that assumes the `.sim.` tandem. I want to think a bit more before just dive into it. If we go about this. We will need to change the (1) wrappers on master including the classroom activity, (2) sonification wrapper but in correlation with new publications, because it break's their api, and (3) the phet-io-website examples when they are updated.

I didn't want to do all of this now, just for a name change, but I'm still open to it.

**Comment by samreid** [comment]
It will be a hassle to change the names of these things now, but it will be much more painful later on.  We should decide what we want to be in stable 1.0s.

**Comment by zepumph** [comment]
Agreed. I guess the cost shouldn't play into the decision to move towards the optimal solution.

**Comment by zepumph** [comment]
What I really meant above is that the cost shouldn't play into deciding what the optimal decision is.

**Comment by zepumph** [comment]
Places that I will not be updating usages of `.sim.` in phetioIDs:
* Sonification wrapper - I will need to coordinate this with @jbphet, when new phet-io sims are released.
* phet-io-website - in `integrate.html` or `customize.html` examples, these too are using fixed versions
* `phet-io/html/pearson/test-invoke-sequence.html`

Everywhere else should be changed. @samreid this one is very tricky, please review carefully. I spent a while testing all that I could think of, as well as searching for usages with `'.*sim\..*' `'.*\.sim\.` and `\.sim\.`. Any recommendations?

**Comment by samreid** [comment]
I tested through instance proxies for faraday's law, and the IDs seem good.  Earlier today I saw a "faradaysLaw.sim.activeProperty not found" in the state wrapper, but I couldn't reproduce the problem again--I may have had an inconsistent state during development.  This seems good to close for now.
